backends/mediatek/neuropilot/preprocess.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2024 MediaTek Inc.
+#
+# Licensed under the BSD License (the "License"); you may not use this file
+# except in compliance with the License. See the license file in the root
+# directory of this source tree for more details.
+
+import contextlib
+import json
+import os
+import shutil
+import subprocess
+import struct
+import tempfile
+import uuid
+
+from pathlib import Path
+from typing import final, List
+from executorch.exir.backend.backend_details import (
+    BackendDetails,
+    ExportedProgram,
+    PreprocessResult,
+)
+from executorch.exir.backend.compile_spec_schema import CompileSpec
+
+import mtk_converter
+import mtk_neuron
+
+SKIP_COMPILE_SPEC_KEYS = {'HighAddr', 'ImportForever'}
+
+
+@final
+class NeuropilotBackend(BackendDetails):
+
+    @classmethod
+    def preprocess(
+        cls,
+        edge_program: ExportedProgram,
+        module_compile_spec: List[CompileSpec]
+    ) -> PreprocessResult:
+
+        header_version = 1
+        num_inputs = len(edge_program.graph_signature.user_inputs)
+        num_outputs = len(edge_program.graph_signature.user_outputs)
+
+        compile_options = ['--arch=mdla5.1', '--relax-fp32']
+        for spec in module_compile_spec:
+            if spec.key in SKIP_COMPILE_SPEC_KEYS:
+                continue
+            value = spec.value.decode('utf-8')
+            compile_options.append(f'--{spec.key}={value}')
+
+        converter = mtk_converter.PyTorchV2Converter.from_exported_program(edge_program)
```

Comment:
It seems like the `mtk_convert` hosted on NP portal is handling go through the graph and process node by node.

From a user perspective, without knowing what ops can be supported by mtk is a bit hard to lower a model to mtk backend. There can be two scenarios:
1. Partially lower the model. While it is the plan to partially lower some oss models to npu, it's a bit hard to tell if it will be supported smoothly as everything is hidden behind `mkt_convert`. 
2. Decompose the not supported ops to the op list supported by npu to fully lower a model to mtk. Similar to scenario 2 (also similar the `decompose_scaled_dot_product_attention` pass), users may want to decompose some non supported ops with the supported ops. It's hard to do so without know what ops are supported.